OpenRA.Mods.Common/Widgets/Logic/Ingame/LeaveMapLogic.cs

```diff
@@ -33,6 +34,11 @@ public LeaveMapLogic(Widget widget, World world, OrderManager orderManager)
 		{
 			this.orderManager = orderManager;
 
+			if (world.LocalPlayer.WinState == WinState.Won)
+			{
+				CampaignProgress.SaveProgress(world.LocalPlayer.Country.Name);
```

Comment:
The progress will only be saved in campaign. We use a flag wich is set by pushing the campaign button in the main menu.